app/services/topic_status_updater.rb

```diff
@@ -90,6 +90,19 @@ def update_read_state_for(status, old_highest_read)
         # actually read the topic
         PostTiming.pretend_read(topic.id, old_highest_read, topic.highest_post_number)
       end
+
+      if status.closed? && status.enabled?
+        sql_query = <<-SQL
+          SELECT DISTINCT post_timings.user_id
+          FROM post_timings
+          JOIN user_options ON user_options.user_id = post_timings.user_id
+          WHERE post_timings.topic_id = :topic_id
+            AND user_options.silence_close_notifications = 't'
+        SQL
+        user_ids = DB.query(sql_query, topic_id: topic.id).map { |u| u.user_id }
```

Comment:
You can just use `DB.query_single` if you only have one column, saves you a map ;)